engine/dlib/src/dlib/http_cache.cpp

```diff
@@ -417,11 +417,12 @@ namespace dmHttpCache
             return RESULT_INVAL;
         }
 
-        uint64_t uri_hash = dmHashString64(uri);
+        size_t uri_length = strlen(uri);
+        uint64_t uri_hash = dmHashBuffer64(uri, uri_length);
 
         HashState64 hash_state;
         dmHashInit64(&hash_state, false);
-        dmHashUpdateBuffer64(&hash_state, uri, strlen(uri));
+        dmHashUpdateBuffer64(&hash_state, uri, uri_length);
```

Comment:
This is a change with little real gain.
Note that you are trading runtime with code size here.

What is the size of the engine before/after your changes?
Note that every bit counts, and it's a huge selling point for our users.

Reply:
Good point. Size increase will not be worth it, so I will close this PR.

I think a better approach to reduce build size, while also avoiding `strlen` calls, would be to use functions like `dmScript::CheckHashOrString()`. All blocks that look something like this:

```cpp
dmhash_t property_hash;
if (dmScript::IsHash(L, 2)) {
    property_hash = dmScript::CheckHash(L, 2);
} else {
    property_hash = dmHashString64(luaL_checkstring(L, 2));
}
```

```cpp
dmhash_t property_id;
if (lua_isstring(L, 2)) {
    property_id = dmHashString64(lua_tostring(L, 2));
} else {
    property_id = dmScript::CheckHash(L, 2);
}
```

Could be replaced by a single function call. 

And if one really wants to, replace instances of `dmHashString64(lua_tostring(L, 2))` with a call to a function that does the `lua_tolstring` to `dmHashBuffer64` dance to avoid `strlen`.